src/OrchardCore/OrchardCore.Email.Core/Services/NullEmailDeliveryService.cs

```diff
@@ -1,8 +1,21 @@
 using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
 
 namespace OrchardCore.Email.Services;
 
 public class NullEmailDeliveryService : IEmailDeliveryService
 {
-    public Task<EmailResult> DeliverAsync(MailMessage message) => Task.FromResult(EmailResult.Success);
+    private readonly ILogger _logger;
+
+    public NullEmailDeliveryService(ILogger<NullEmailDeliveryService> logger)
+    {
+        _logger = logger;
+    }
+
+    public async Task<EmailResult> DeliverAsync(MailMessage message)
+    {
+        _logger.LogWarning("No email delivery service is configured. Please enable an actual implementation so the email can send.");
+
+        return await Task.FromResult(EmailResult.Success);
```

Comment:
Shouldn't this be failure?

Reply:
I don't think while it uses a null object pattern

Reply:
I mean, shouldn't we return `EmailResult.Failure` here, given that no e-mail was actually sent?

Reply:
This something like logger, when `NullLogger` doesn't log anything

Reply:
Well, this does log :).

Reply:
Done